content/api/commands/origin.md

```diff
@@ -0,0 +1,467 @@
+
+Pass in an options object to control the behavior of `cy.origin()`.
+
+| option | description                                                                                                                                                                                                    |
+| ------ | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| args   | Plain JavaScript object which will be serialized and sent from the primary origin to the secondary origin, where it will be deserialized and passed into the callback function as its first and only argument. |
+
+<Alert type="warning">
+
+The `args` object is the **only** mechanism via which data may be injected into
+the callback, the callback is **not** a closure and does not retain access to
+the JavaScript context in which it was declared.
+
+</Alert>
+
+**<Icon name="angle-right"></Icon> callbackFn** **_(Function)_**
+
+The function containing the commands to be executed in the secondary origin.
+
+This function will be stringified, sent to the Cypress instance in the secondary
+origin and evaluated. If the `args` option is specified, the deserialized args
+object will be passed into the function as its first and only argument.
+
+There are a number of limitations placed on commands run inside the callback,
+please see [Callback restrictions](#Callback-restrictions) section below for a
+full list.
+
+The function's return value is ignored. ???
+
+### Yields [<Icon name="question-circle"/>](/guides/core-concepts/introduction-to-cypress#Subject-Management)
+
+- `cy.origin()` yields `null`.
+- `cy.origin()` cannot be chained further.
+
+## Examples
+
+### Using dynamic data in a secondary origin
+
+Callbacks are executed inside an entirely separate instance of Cypress, so
+arguments must be transmitted to the other instance by means of serialization &
+deserialization. The interface for this mechanism is the `args` option.
+
+```js
+const sentArgs = { password: 'P@55w0rd!!!1!?!111one' }
+cy.origin(
+  'supersecurelogons.com',
+  // Send the args here...
+  { args: sentArgs },
+  // ...and receive them at the other end here!
+  (receivedArgs) => {
+    const { password } = receivedArgs
+    cy.visit('/login')
+    cy.get('input#password').type(password)
+    cy.contains('button', 'Login').click()
+  }
+)
+```
+
+Note: You can just replace `sentArgs` and `receivedArgs` with `args` if you
+want, the naming in this example is purely for clarity.
+
+### Navigating to secondary origin with cy.visit
+
+When navigating to a secondary origin using `cy.visit()`, it is essential to
+trigger the navigation **after** entering the origin callback, otherwise a
+cross-domain error will be thrown.
+
+```js
+// Do things in primary domain...
+
+cy.origin('acme.com', () => {
+  // Visit https://www.acme.com/history/founder
```

Comment:
The subdomain won't be added since it wasn't passed to `cy.origin`. In this case, the relative url would expand to: `https://acme.com/history/founder`.